src/layout/RepeatingGroup/Summary2/RepeatingGroupSummary.tsx

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+
+import { ErrorMessage } from '@digdir/designsystemet-react';
+import { ExclamationmarkTriangleIcon } from '@navikt/aksel-icons';
+import cn from 'classnames';
+
+import { Label } from 'src/components/label/Label';
+import { Lang } from 'src/features/language/Lang';
+import { useUnifiedValidationsForNode } from 'src/features/validation/selectors/unifiedValidationsForNode';
+import { validationsOfSeverity } from 'src/features/validation/utils';
+import { useRepeatingGroupRowState } from 'src/layout/RepeatingGroup/RepeatingGroupProviders/RepeatingGroupContext';
+import classes from 'src/layout/RepeatingGroup/Summary2/RepeatingGroupSummary.module.css';
+import { SingleValueSummary } from 'src/layout/Summary2/CommonSummaryComponents/SingleValueSummary';
+import { BaseLayoutNode } from 'src/utils/layout/LayoutNode';
+import { Hidden } from 'src/utils/layout/NodesContext';
+import { useNodeItem } from 'src/utils/layout/useNodeItem';
+import type { CompTypes } from 'src/layout/layout';
+import type { AnyComponent } from 'src/layout/LayoutComponent';
+import type { Summary2Props } from 'src/layout/Summary2/SummaryComponent2/types';
+
+export const RepeatingGroupSummary = ({ target, isCompact, overrides }: Summary2Props<'RepeatingGroup'>) => {
+  const validations = useUnifiedValidationsForNode(target);
+  const { visibleRows } = useRepeatingGroupRowState();
+  const rowsToDisplaySet = new Set(visibleRows.map((row) => row.uuid));
+  const rows = useNodeItem(target, (i) => i.rows).filter((row) => row && rowsToDisplaySet.has(row.uuid));
+  const errors = validationsOfSeverity(validations, 'error');
+  const title = useNodeItem(target, (i) => i.textResourceBindings?.title);
+  const isNested = target.parent instanceof BaseLayoutNode;
+
+  if (rows.length === 0) {
+    return (
+      <SingleValueSummary
+        componentNode={target}
+        title={title}
+      />
+    );
+  }
+
+  return (
+    <div
+      className={cn({ [classes.nestedRepeatingGroupSummaryWrapper]: isNested })}
+      data-testid={'summary-repeating-group-component'}
+    >
+      <Label
+        node={target}
+        renderLabelAs='span'
```

Comment:
Fieldset should be used when there are form elements that are grouped together. Here we currently do not have any form elements due to how we render the summary components. 

I think we should look at rendering the summary components as `readonly` input fields where we can first to make them semantic and accessible (This might not be possible for all of the summary components like multiple-select). After that it would make sense to change this to a fieldset with a legend.

Reply:
I see! I'm not as into all the accessibility stuff as you are, but making all these summaries into `readOnly` inputs sure sounds like the opposite of previous recommendations.. :face_with_peeking_eye: How it behaves in screen readers, etc, is the most important thing, I guess.